bindings/matrix-sdk-ffi/src/lib.rs

```diff
@@ -18,6 +18,8 @@ mod platform;
 mod room;
 mod room_directory_search;
 mod room_info;
+
+#[allow(deprecated)]
 mod room_list;
```

Comment:
How many uses of the deprecated functions are there in this module? Can we add the guard on them one by one instead, or would that be too many? If too many, can we put the allow guard in the module itself, instead of here?

Reply:
When I tried using the allow guard in the function it didn't take effect (I still had the same compiler error message), I had to place it in the module for it to work. I suspect it has something to do with the FFI generated code, but I'm not sure.

Reply:
Ah yeah, expanding the macro could tell us whether it's possible to put it on a slightly more specialized location, e.g. the `impl` block itself, or if it's using it in code generated elsewhere.

Reply:
Adding it to the impl block fails too. I've only been able to fix it by adding it in the `mod` declaration.

Reply:
When I add it to the function, the expanded macro looks like this:

```rust
#[doc = " Builds a `Room` FFI from an invited room without initializing its"]
#[doc = " internal timeline."]
#[doc = ""]
#[doc = " An error will be returned if the room is a state different than invited."]
#[doc = ""]
#[doc = " ⚠\u{65039} Holding on to this room instance after it has been joined is not"]
#[doc = " safe. Use `full_room` instead."]
#[allow(deprecated)]
#[deprecated(note = "Please use `preview_room` instead.")]
fn invited_room(&self) -> Result<Arc<Room>, RoomListError> {
    if !match (self.membership()) {
        Membership::Invited => true,
        _ => false
    } {
        return Err(RoomListError::IncorrectRoomMembership {
            expected: (<[_]>::into_vec(
                #[rustc_box]
                ::alloc::boxed::Box::new([Membership::Invited])
            )),
            actual: self.membership(),
        });
    }
    Ok(Arc::new(Room::new(self.inner.inner_room().clone())))
}
```

And if I add it to the impl block it looks like:

```rust
#[uniffi::export(async_runtime = "tokio", )]
#[allow(deprecated)]
impl RoomListItem {
    ...
```

But clippy still complains in both cases.

Reply:
Yeah, it's the call sites that need the `allow(deprecated)` guard, not the function itself; but it might be that it's in code generated by uniffi, that's outside our control, so we might need to resort to the mod-level guard indeed.

Reply:
Forgot to comment here: it's indeed an unmangled free-function generated in the same module, that's calling the `invited_room` method, so welp :shrug: Module-level guard it is. I wondered if we could get rid of the entire `invited_room` method altogether, but I don't want to add more API changes to the other side of this PR.